//
// Created by ada on 2021/1/23.
//

syntax = "proto3";

package comm;

message UserRequest {
  bytes payload = 1;
  bytes digest = 2;
}

message ClientToUserResponse{
  uint32 err_number = 1;
}

message EpochRequest{
  uint32 request_batch_size = 1;
  bytes batch_hash = 2;
}

message EpochResponse{
  uint32 epoch = 1;
  EpochRequest request = 2;
  uint32 nonce = 3;
  bytes signature = 4;
}

// order is needed for transactions
message TransactionsWithProof {
  bytes proof = 1;
  repeated bytes transaction = 2;
}

message ProxyBroadcast {
  uint32 epoch = 1;
  repeated TransactionsWithProof set = 2;
}

message SignatureExchange {
  uint32 epoch = 1;
  bytes signature = 2;
}